src/java.management/share/classes/javax/management/ObjectName.java

```diff
@@ -2020,9 +2020,11 @@ private final boolean matchDomains(ObjectName name) {
             // wildmatch domains
             // This ObjectName is the pattern
             // The other ObjectName is the string.
-            return Util.wildmatch(name.getDomain(),getDomain());
+            return Util.wildmatch(name._canonicalName, _canonicalName,
+                   0, 0, name.getDomainLength(), getDomainLength());
```

Comment:
This looks wrong. Look at the parameters in wildmatch: 
```
wildmatch(final String str, final String pat,
                   int stri, final int strend, int pati, final int patend)
```

What tests did you run to validate your changes? If none of the ObjectName tests fail with your changes you will need to add a new test that catches the error.

Reply:
Thank you again for review and finding.
Current tests do not catch such issue, so I extended ApplyWildcardTest.java with domain patterns. The test passed on the original and new code, fails with incorrect Util.wildmatch parameters